api/src/Entity/DayResponsible.php

```diff
@@ -50,7 +50,7 @@ class DayResponsible extends BaseEntity implements BelongsToCampInterface {
     /**
      * The person that is responsible. Must belong to the same camp as the day's period.
      *
-     * @ORM\ManyToOne(targetEntity="CampCollaboration", inversedBy="dayResponsibles")
+     * @ORM\ManyToOne(targetEntity="CampCollaboration", inversedBy="dayResponsibles", cascade={"persist"})
```

Comment:
This one we shouldn't do. We never want to add (persist) a new CampCollaboration through the DayResponsible endpoint.

Reply:
True. But currently the bigger problem is, Doctrine complains that Category#camp is not persisted, and I have no idea why. It's camp1 which seems to not be included in the inserted entities in the UOW, and I'm not sure why not.

Reply:
You mean this one:
https://github.com/ecamp/ecamp3/runs/4788619335?check_suite_focus=true#step:6:3417

Well it seems the camp is removed successfully (both from DB and EntityManager).

But then the camp is found again via the category (which is not removed from the EntityManager) and triggered through this flush:
https://github.com/ecamp/ecamp3/pull/2391/files#diff-8c0080302b85a38645ffef7a002c65a761b3c4309abfabbf4a7bdee12e2cb201R32

We could probably get rid of the message through `cascade={"remove"}` to ensure category entities are removed from the EntityManager together with the camp. But that's not really what we want, I guess, because we already have the more performant `onDelete="cascade"` on the DB.

So question would be whether this flush is correct?

Reply:
I think this one is the exact problem:
https://github.com/doctrine/orm/issues/9054

So can we delete the extra flush? Or is this needed for other operations?